pkg/glusterfs/api/types.go

```diff
@@ -330,6 +330,16 @@ func (volExpandReq VolumeExpandRequest) Validate() error {
 	)
 }
 
+type VolumeCloneRequest struct {
+	Name string `json:"name,omitempty"`
```

Comment:
The volumeClone request can also have `size` field in it and also `cluster` field , I am not sure cloning to a different cluster is possible , if not we dont need cluster field. Also this could have an optional `snap` field, so if user specify the `snap` name, we could take clone from it. @nixpanic @phlogistonjohn comments?

Reply:
It is not possible to do clones from one cluster to an other. Even the clones use the same block devices as the original bricks. This is just how Gluster implements volume cloning.

The extended snapshot API that is not included in the PR, offers the full create-snapshot and clone-volume-from-snapshot options. It is still planned to add that, but at a later stage. See the design from commit 353e8bdb8 for all operations.